### PR TITLE
Modernization of the code (2023)

### PR DESCRIPTION
**Pull request by dbaty**
Commits should be reviewed one by one. Summary:

- drop support of old Python and Django;
- add support of recent versions of Python and Django;
- move to a recent version of Sphinx
- run quality check on both minimum and maximum supported versions of Python and Django
- etc.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -3,7 +3,7 @@ include CONTRIBUTING.rst
 include HISTORY.rst
 include LICENSE
 include README.rst
-recursive-include cid *.html *.png *.gif *js *.css *jpg *jpeg *svg *py
+recursive-include src/cogite *.html *.png *.gif *js *.css *jpg *jpeg *svg *py
```
Comment (RemiCardona) [inline MANIFEST.in:6]: pretty sure `cogite` isn't intended :)

Also, shouldn't there be a modification in `setup.py` to support the move to `src/`?

**Review by RemiCardona — APPROVED** [review]

File: MANIFEST.in (hunk 1 above)
Comment (dbaty) [reply MANIFEST.in:6]: I did not copy and paste this line from another project. No. Not at all. I would never do that.

> Shouldn't there be a modification [...] to support the move to src/?

Good catch. I did that in [another commit](https://github.com/Polyconseil/django-cid/pull/62/commits/af730d708dcb70b0adcc04749371a79eaba0ec04) later. So... this commit is not atomic! I'll probably go to hell for that.

File: docs/Makefile [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-SPHINXOPTS    =
-SPHINXBUILD   = sphinx-build
-SPHINXPROJ    = DjangoCorrelationId
+# You can set these variables from the command line, and also
+# from the environment for the first two.
+SPHINXOPTS    ?=
+SPHINXBUILD   ?= sphinx-build
 SOURCEDIR     = .
-BUILDDIR      = ../build/sphinx
+BUILDDIR      = _build
```
Comment (sandre35) [inline docs/Makefile:9]: Isn't there a conflict with `exclude_patterns` which includes the new BUILDDIR path ?
https://github.com/Polyconseil/django-cid/pull/62/commits/419cbda0204d0f6ba5177df5449e45c09ab564c1#diff-85933aa74a2d66c3e4dcdf7a9ad8397f5a7971080d34ef1108296a7c6b69e7e3R45

File: sandbox/testapp/migrations/0001_initial.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Generated by Django 2.1.1 on 2018-09-03 13:13
+# Generated by Django 3.1 on 2023-07-28 10:00
```
Comment (sandre35) [inline sandbox/testapp/migrations/0001_initial.py:1]: Is it necessary ? :) 
Django keeps compatibility of migration files with previous version.

File: docs/Makefile (hunk 2 above)
Comment (dbaty) [reply docs/Makefile:9]: If I remember correctly, this is the default Sphinx configuration. The Makefile says that we build in a `_build/` directory, and the Sphinx configuration says that Sphinx should not look for source files in that directory. It seems ok to me. Did I miss something?

File: sandbox/testapp/migrations/0001_initial.py (hunk 3 above)
Comment (sandre35) [reply sandbox/testapp/migrations/0001_initial.py:1]: Ahah yes no problem about that :) 

I thought that Django supported migration files with previous Django version. Apparently it is not the case according to the [official documentation](https://docs.djangoproject.com/en/4.2/topics/migrations/#supporting-multiple-django-versions). Recreating migration files was the correct decision ^^

File: docs/Makefile (hunk 2 above)
Comment (sandre35) [reply docs/Makefile:9]: Okay, I misunderstood the behaviour of `exclude_patterns`. All good for me :+1:

**Review by sandre35 — APPROVED** [review]

**Pull request merged**